packages/shorebird_cli/lib/src/commands/doctor_command.dart

```diff
@@ -36,33 +43,74 @@ class DoctorCommand extends ShorebirdCommand with ShorebirdVersionMixin {
 
   @override
   Future<int> run() async {
+    final shouldFix = results['fix'] == true;
+
     logger.info('''
 
 Shorebird v$packageVersion
 Shorebird Engine • revision ${ShorebirdEnvironment.shorebirdEngineRevision}''');
 
-    var numIssues = 0;
+    final allIssues = <ValidationIssue>[];
+    final allFixableIssues = <ValidationIssue>[];
     for (final validator in validators) {
       final progress = logger.progress(validator.description);
       final issues = await validator.validate(process);
-      numIssues += issues.length;
       if (issues.isEmpty) {
         progress.complete();
-      } else {
-        progress.fail();
+        continue;
+      }
 
-        for (final issue in issues) {
-          logger.info('  ${issue.displayMessage}');
+      final fixableIssues = issues.where((issue) => issue.fix != null);
+      var unresolvedIssues = issues;
+      if (fixableIssues.isNotEmpty) {
+        if (shouldFix) {
+          // If --fix flag was used and there are fixable issues, fix them.
+          progress.update('Fixing');
+          for (final issue in fixableIssues) {
+            await issue.fix!();
+          }
+
+          // Re-run the validator to see if there are any remaining issues that
+          // we couldn't fix.
+          unresolvedIssues = await validator.validate(process);
```

Comment:
Won't this result in potentially lots of validations running twice? In what circumstances would fix complete successfully without actually fixing the issue?

Reply:
Wouldn't fix throw in those cases? It just feels weird that we don't trust a completed fix to fix the issue and instead need to re-run the validator. Why is it not enough to run fix in a `try/catch` and if it throws mark it as failed otherwise assume it succeeded?

Reply:
Maybe! It may also be possible that fixing one issue allows the validator to find an error further along in the validation.